src/frontend/src/optimizer/plan_node/generic/scan.rs

```diff
@@ -276,18 +291,21 @@ impl Scan {
         // required columns, i.e., the mapping from operator_idx to table_idx.
 
         let mut required_col_idx = output_col_idx.clone();
-        let predicate_col_idx = predicate.collect_input_refs(table_desc.columns.len());
+        let predicate_col_idx = predicate.collect_input_refs(table_catalog.columns().len());
         predicate_col_idx.ones().for_each(|idx| {
             if !required_col_idx.contains(&idx) {
                 required_col_idx.push(idx);
             }
         });
 
+        let table_desc = Rc::new(table_catalog.table_desc());
```

Comment:
We can have everything reference `table_catalog`, but that requires a fair amount of refactoring. So do that in separate PR.